src/services/authorizedItem.service.getPropertiesForItem.test.ts

```diff
@@ -0,0 +1,443 @@
+
+    jest
+      .spyOn(itemMembershipRepository, 'getInherited')
+      .mockImplementation(
+        async () => ({ permission: 'anything' }) as unknown as ItemMembershipWithItemAndAccount,
+      );
+
+    // any other member shouldn't access
+    await expect(
+      authorizationService.getPropertiesForItem(MOCK_DB, {
+        permission: PermissionLevel.Admin,
+        actor: MEMBER,
+        item: ITEM,
+      }),
+    ).rejects.toBeInstanceOf(Error);
+  });
+
+  it.each([
+    // no membership
+    {
+      permission: PermissionLevel.Read,
+      membership: null,
+      rejects: MemberCannotAccess,
+    },
+    {
+      permission: PermissionLevel.Write,
+      membership: null,
+      rejects: MemberCannotAccess,
+    },
+    {
+      permission: PermissionLevel.Admin,
+      membership: null,
+      rejects: MemberCannotAccess,
+    },
+    // shared item with read permission
+    {
+      permission: PermissionLevel.Read,
+      membership: readMembership,
+      returns: readMembership,
+    },
+    {
+      permission: PermissionLevel.Write,
+      membership: readMembership,
+      rejects: MemberCannotWriteItem,
+    },
+    {
+      permission: PermissionLevel.Admin,
+      membership: readMembership,
+      rejects: MemberCannotAdminItem,
+    },
+    // shared item with write permission
+    {
+      permission: PermissionLevel.Read,
+      membership: writeMembership,
+      returns: writeMembership,
+    },
+    {
+      permission: PermissionLevel.Write,
+      membership: writeMembership,
+      returns: writeMembership,
+    },
+    {
+      permission: PermissionLevel.Admin,
+      membership: writeMembership,
+      rejects: MemberCannotAdminItem,
+    },
+    // shared item with admin permission
+    {
+      permission: PermissionLevel.Read,
+      membership: adminMembership,
+      returns: adminMembership,
+    },
+    {
+      permission: PermissionLevel.Write,
+      membership: adminMembership,
+      returns: adminMembership,
+    },
+    {
+      permission: PermissionLevel.Admin,
+      membership: adminMembership,
+      returns: adminMembership,
+    },
+  ])(
+    'request $permission for private item',
+    async ({ permission, membership, returns, rejects }) => {
+      jest.spyOn(itemVisibilityRepository, 'getByItemPath').mockResolvedValue([]);
+      jest
+        .spyOn(itemMembershipRepository, 'getInherited')
+        .mockImplementation(async () => membership);
+
+      await authorizationService
+        .getPropertiesForItem(MOCK_DB, {
+          permission,
+          actor: MEMBER,
+          item: ITEM,
+        })
+        .then(({ itemMembership: result }) => {
+          if (rejects) {
+            throw new Error('should throw');
+          }
+          expect(result).toEqual(returns);
+        })
+        .catch((e) => {
+          if (returns) {
+            throw new Error('should return');
+          }
+          expect(e).toBeInstanceOf(rejects);
+        });
```

Comment:
I admit that this is a clever way to do it, but I do not think that it is easy to understand or update in case we need to change. Also I am not a fan to throwing errors inside tests in order to make the test fail ? that seems counter-productive.